### ISSUE TITLE
transactional-update.service should not run if a reboot is already scheduled

### ISSUE THREAD

**Issue by RlndVt** [openSUSE/transactional-update#77]
If transactional-update is used to install a package with a reboot request e.g.: `transactional-update reboot pkg in $PACKAGE`, the transactional-update.service should (check with rebootmgr, see that a reboot is scheduled, and) not attempt to update your packages as that will 'override' the package install. 

Usecase is if you want to install a package but do not want to take your server down until the scheduled moment.

Workaround is stopping the `transactional-update.timer` service after installing a package.

**Comment by aplanas** [comment]
Another option is to use `--continue` to do a new snapshot based on the one yet-to-be-activated. IMHO this should be the default behavior?

**Comment by laenion** [comment]
Agreed, this looks like a good idea.

**Comment by thkukuk** [comment]
That's a bad idea:

- Create a new snapshot which is broken
- Boot that, do a rollback
- next transactional-update.timer run will use the broken snapshot as base

=> No self-healing is possible anymore.

If you do manual updates and request a reboot, you have to disable the timer so long or make sure the reboot is done upfront. Everything else doesn't make any sense.

**Comment by laenion** [comment]
I edited my first post, I meant the original request.

There's one disadvantage of this approach though: Let's say that the system is configured to reboot only once a week. In that case any further updates would be prevented after the first update, so the snapshot would probably not be up to date any more when the reboot is finally triggered.

**Comment by RlndVt** [comment]
@thkukuk To clarify, do you think the suggested check would be a good addition?

**Comment by thkukuk** [comment]
I don't think so, as you don't know who did request the reboot and why. rebootmgr is by no means coupled or only useable by transactional-update. So if somebody requests a reboot, the reboot would be done without the latest patches applied and a second reboot would be necessary, even if this all could be done in one step. In worst case updates will not applied for several days.
I really have no good idea how to solve that except reboot yourself immediately after installing the additional package.
